mono/metadata/icall.c

```diff
@@ -7334,6 +7334,55 @@ ves_icall_System_IO_get_temp_path (MonoError *error)
 	return mono_string_new_handle (mono_domain_get (), g_get_tmp_dir (), error);
 }
 
+#if defined(ENABLE_MONODROID) || defined(ENABLE_MONOTOUCH)
+
+gpointer
+CreateZStream (gint32 compress, MonoBoolean gzip, gpointer feeder, gpointer data);
+
+gpointer
+ves_icall_System_IO_Compression_DeflateStreamNative_CreateZStream (gint32 compress, MonoBoolean gzip, gpointer feeder, gpointer data)
+{
+	return CreateZStream (compress, gzip, feeder, data);
+}
+
+gint32
+CloseZStream (gpointer stream);
+
+gint32
+ves_icall_System_IO_Compression_DeflateStreamNative_CloseZStream (gpointer stream)
```

Comment:
Could these all be PInvokes? is there any reason to add icalls?

Reply:
p/invoke on XA/XI is inherently broken because `__Internal` points to `libmonodroid` (and XI's equivalent), not `libmonosgen-2.0`. So `InternalCall` are just the sure way to hit the right target.